pyfixest/estimation/decomposition.py

```diff
@@ -0,0 +1,390 @@
+    mask: np.ndarray = field(init=False)
+    mediator_names: list[str] = field(init=False)
+    X_dict: dict[Any, Any] = field(init=False, default_factory=dict)
+    X1_dict: dict[Any, Any] = field(init=False, default_factory=dict)
+    X2_dict: dict[Any, Any] = field(init=False, default_factory=dict)
+    Y_dict: dict[Any, Any] = field(init=False, default_factory=dict)
+
+    def __post_init__(self):
+        param_idx = self.coefnames.index(self.param)
+        self.intercept_idx = self.coefnames.index("Intercept")
+        self.coefnames_no_intercept = self.coefnames[~self.intercept_idx]
+        self.mask = np.ones(len(self.coefnames), dtype=bool)
+        self.mask[param_idx] = False
+
+        self.mediator_names = [
+            name for name in self.coefnames if self.param not in name
+        ]
+        self.intercept_in_mediator_idx = self.mediator_names.index("Intercept")
+
+        # Handle clustering setup if cluster_df is provided
+        if self.cluster_df is not None:
+            self.unique_clusters = self.cluster_df.unique()
+            self.cluster_dict = {
+                cluster: self.cluster_df[self.cluster_df == cluster].index
+                for cluster in self.unique_clusters
+            }
+        else:
+            self.unique_clusters = None
+            self.cluster_dict = None
+
+        if self.combine_covariates is None:
+            self.combine_covariates_dict = {
+                x: [x] for x in self.mediator_names if x != "Intercept"
+            }
+        else:
+            self.combine_covariates_dict = self.combine_covariates
+
+        self._check_combine_covariates()
+
+    def _check_combine_covariates(self):
+        # Check that each value in self.combine_covariates_dict is in self.mediator_names
+        for _, values in self.combine_covariates_dict.items():
+            if not isinstance(values, list):
+                raise TypeError("Values in combine_covariates_dict must be lists.")
+            for v in values:
+                if v not in self.mediator_names:
+                    raise ValueError(f"{v} is not in the mediator names.")
+
+        # Check for overlap in values between different keys
+        all_values = {
+            k: set([v] if isinstance(v, str) else v)
+            for k, v in self.combine_covariates_dict.items()
+        }
+        for key1, values1 in all_values.items():
+            for key2, values2 in all_values.items():
+                if key1 != key2 and values1 & values2:
+                    overlap = values1 & values2
+                    raise ValueError(f"{overlap} is in both {key1} and {key2}.")
+
+    def fit(self, X: np.ndarray, Y: np.ndarray, store: bool = True):
+        "Fit Linear Mediation Model."
+        if store:
+            self.X = X
+            self.N = X.shape[0]
+
+            self.X1 = self.X[:, ~self.mask]
+            self.X1 = np.concatenate([np.ones((self.N, 1)), self.X1], axis=1)
```

Comment:
q (for my own curiosity): any benefits to `np.concatenate` over `np.vstack` or even more concise `np.c_[arr1, arr2]` ? I use the latter

Reply:
Not that I am aware of, just habit =)